src/components/UserSelector.tsx

```diff
@@ -1,51 +1,50 @@
-import React, { useContext, useEffect, useState } from 'react';
+/* eslint-disable consistent-return */
+import {
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
 import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { actions as authorActions } from '../features/authorSlice';
+import { fetchUsers } from '../features/userSlice';
 
-type Props = {
-  value: User | null;
-  onChange: (user: User) => void;
-};
-
-export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
-  value: selectedUser,
-  onChange,
-}) => {
-  // `users` are loaded from the API, so for the performance reasons
-  // we load them once in the `UsersContext` when the `App` is opened
-  // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
+export const UserSelector = () => {
+  const dispatch = useAppDispatch();
+  const { users } = useAppSelector((state) => state.users);
+  const { author } = useAppSelector((state) => state.author);
   const [expanded, setExpanded] = useState(false);
 
+  const handleSelectAuthor = useCallback((user: User) => {
+    dispatch(authorActions.setAuthor(user));
+  }, []);
+
+  useEffect(() => {
+    dispatch(fetchUsers());
+  }, []);
+
   useEffect(() => {
     if (!expanded) {
       return;
     }
 
-    // we save a link to remove the listener later
     const handleDocumentClick = () => {
-      // we close the Dropdown on any click (inside or outside)
-      // So there is not need to check if we clicked inside the list
       setExpanded(false);
     };
 
     document.addEventListener('click', handleDocumentClick);
 
-    // eslint-disable-next-line consistent-return
     return () => {
       document.removeEventListener('click', handleDocumentClick);
     };
-  // we don't want to listening for outside clicks
-  // when the Dopdown is closed
   }, [expanded]);
 
   return (
     <div
-      data-cy="UserSelector"
-      className={classNames('dropdown', { 'is-active': expanded })}
+      data-cy="authorelector"
```

Comment:
```suggestion
      data-cy="UserSelector"
```
looks like it is reason of failed tests
![Screenshot from 2023-09-26 16-04-46](https://github.com/mate-academy/react_redux-list-of-posts/assets/33029864/5065bb23-5130-4899-aa7e-e68ea72ba89a)